src/hashTable.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const ARBITRARY_ZIP_CODE_HASH = 94607
+const LENGTH_OF_ALPHABET = 26
+const node = ( key, value ) => {
+  let obj = {}
+  obj[ key ] = value
+  return obj
+}
+
+export default class HashTable {
+
+  constructor() {
+    this.length = 0
+    this.collection = {}
+  }
+
+  put( key, value ) {
+
+    if ( !value ) {
+      throw new Error( 'Missing necessary parameter(s)' )
+    }
+    if ( typeof key !== 'string' ) {
+      throw new Error( 'Key must be a string' )
+    }
+
+    let index = HashTable.hash( key ) % LENGTH_OF_ALPHABET
+    let newNode = node( key, value )
+
+    if ( this.collection[index] ) {
+      this.collection[index].push( newNode )
+    }
+    else {
+      this.collection[index] = [ newNode ]
+    }
+
+    this.length++
+
+  }
+
+  get( key ) {
+
+    if ( !key ) {
+      throw new Error( 'No key given' )
+    }
+    if ( typeof key !== 'string' ) {
+      throw new Error( 'Key given must be a string' )
+    }
+
+    let index = HashTable.hash( key ) % LENGTH_OF_ALPHABET
+
+    if( this.collection[ index ] ) {
+      let keyFilter = this.collection[ index ].filter( node =>
+        Object.keys( node )[0] === key )
+      if( keyFilter.length > 0 ) {
+        return Object.values(keyFilter[0])[0]
+      }
+      else {
+        return null
+      }
+    }
+    else {
+      return null
+    }
+
+  }
+
+  contains( key ) {
+
+    if ( !key ) {
+      throw new Error( 'No key given' )
+    }
+    if ( typeof key !== 'string' ) {
+      throw new Error( 'Key given must be a string' )
+    }
```

Comment:
these error checks are repeated. Consider moving them into a 'validateKey' function defined at the bottom of this file?